controller/pgcluster/pgclustercontroller.go

```diff
@@ -314,3 +324,131 @@ func addIdentifier(clusterCopy *crv1.Pgcluster) {
+	// first, get a list of all of the available deployments so we can properly
+	// mount the tablespace PVCs after we create them
+	// NOTE: this will also get the pgBackRest deployments, but we will filter
+	// these out later
+	selector := fmt.Sprintf("%s=%s,%s=%s", config.LABEL_VENDOR, config.LABEL_CRUNCHY,
+		config.LABEL_PG_CLUSTER, newCluster.Name)
+
+	deployments, err := kubeapi.GetDeployments(c.PgclusterClientset, selector, newCluster.Namespace)
+
+	if err != nil {
+		return err
+	}
+
+	// now get the instance names, which will make it easier to create all the
+	// PVCs
+	instanceNames := []string{}
+
+	for _, deployment := range deployments.Items {
+		labels := deployment.ObjectMeta.GetLabels()
+
+		// get the name of the PostgreSQL instance. If the "deployment-name"
+		// label is not present, then we know it's not a PostgreSQL cluster.
+		// Otherwise, the "deployment-name" label doubles as the name of the
+		// instance
+		if instanceName, ok := labels[config.LABEL_DEPLOYMENT_NAME]; ok {
+			log.Debugf("instance found [%s]", instanceName)
+
+			instanceNames = append(instanceNames, instanceName)
+		}
+	}
+
+	// iterate through the the tablespace mount map that is present in the new
+	// cluster. Any entry that is not in the old cluster, create PVCs
+	newTablespaces := map[string]crv1.PgStorageSpec{}
+
+	for tablespaceName, storageSpec := range newCluster.Spec.TablespaceMounts {
+		// if the tablespace does not exist in the old version of the cluster,
+		// then add it in!
+		if _, ok := oldCluster.Spec.TablespaceMounts[tablespaceName]; !ok {
+			log.Debugf("new tablespace found: [%s]", tablespaceName)
+
+			newTablespaces[tablespaceName] = storageSpec
+		}
+	}
+
+	// now we can start creating the new tablespaces! First, create the new
+	// PVCs. The PVCs are created for each **instance** in the cluster, as every
+	// instnace needs to have a distinct PVC for each tablespace
+	for tablespaceName, storageSpec := range newTablespaces {
+		for _, instanceName := range instanceNames {
+			// get the name of the tablespace PVC for that instance
+			tablespacePVCName := operator.GetTablespacePVCName(instanceName, tablespaceName)
+
+			log.Debugf("creating tablespace PVC [%s] for [%s]", tablespacePVCName, instanceName)
+
+			// and now create it! If it errors, we just need to return, which
+			// potentially leaves things in an inconsistent state, but at this point
+			// only PVC objects have been created
+			if err := clusteroperator.CreateTablespacePVC(c.PgclusterClientset, newCluster.Namespace, newCluster.Name,
+				tablespacePVCName, &storageSpec); err != nil {
+				return err
+			}
+		}
+	}
+
+	// now the fun step: update each deployment with the new volumes
+	for _, deployment := range deployments.Items {
+		labels := deployment.ObjectMeta.GetLabels()
+
+		// same deal as before: if this is not a PostgreSQL instance, skip it
+		instanceName, ok := labels[config.LABEL_DEPLOYMENT_NAME]
```

Comment:
Should we filter these once up top and reuse in this step? No strong opinions either way, just seems like a potential optimization that may be worth the effort. :man_shrugging:

Reply:
Yeah, I had thought about that. There was a reason I had not, but I can't recall what it was and it may be moot at this point.

Reply:
Ah ok. So we had needed the instance names in the step before. I thought the tradeoff (adding in the apps dependency, storing the deployments in an additional array) vs. the actual difference in size was not incrementally worth it.